pkg/valueset/config.go

```diff
@@ -0,0 +1,118 @@
+package valueset
+
+import (
+	"database/sql"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/cloudprivacylabs/lsa/layers/cmd/cmdutil"
+	"gopkg.in/yaml.v2"
+)
+
+type Config struct {
+	ValuesetDBs []ValuesetDB
+}
+
+/*
+	Databases []struct {
+		postgresql.Database `json:"database" yaml:"database"`
+	} `json:"databases" yaml:"databases"`
+*/
+
+type ValuesetDB interface {
+	ValueSetLookup(string, map[string]string) (map[string]string, error)
+	GetTableIds() []string
+}
+
+var DatabaseFactory = map[string]func(interface{}) (ValuesetDB, error){
+	"pgx": NewPostgresqlDataStore,
+}
+
+func RegisterDB(name string, fn func(interface{}) (ValuesetDB, error)) {
+	DatabaseFactory[name] = fn
+}
+
+func init() {
+	RegisterDB("pgx", NewPostgresqlDataStore)
+}
+
+type PostgesqlDataStore struct {
+	dsn string
+	db  *sql.DB
+}
+
+func (pgx PostgesqlDataStore) ValueSetLookup(s string, m map[string]string) (map[string]string, error) {
+	return map[string]string{}, nil
+}
+func (pgx PostgesqlDataStore) GetTableIds() []string {
+	return []string{}
+}
+func NewPostgresqlDataStore(value interface{}) (ValuesetDB, error) {
+	return PostgesqlDataStore{}, nil
+}
+
+type unmarshalConfig struct {
+	Databases map[string]interface{} `json:"databases" yaml:"databases"`
+}
+
+// var m = map[string]func(...interface{}){"foo": foo, "bar": bar}
+// var c = map[string]int{"foo": 1, "bar": 2}
+
+// func (cfg Config) Close() {
+// 	for _, db := range cfg.Databases {
+// 		db.Database.Driver.Close()
+// 	}
+// }
+
+func parseYAML(filename string) (Config, error) {
```

Comment:
refactor this:
  UnmarshalConfig(map[string]interface(), env map[string]string) -- this should have the main logic and unmarshal DBs
  LoadConfig(fileName, env map[string]stirng) -- use the lsa cmdutil ReadJSONorYAML, and call UnmarshalConfig